sqlmesh/core/macros.py

```diff
@@ -177,32 +182,23 @@ def _transform_node(node: exp.Expression) -> exp.Expression:
                     changed = True
                     node.set("this", self.jinja_env.from_string(node.this).render())
                 return node
-            return node
-
-        expression = expression.transform(_transform_node)
-
-        def evaluate_macros(
-            node: exp.Expression,
-        ) -> exp.Expression | t.List[exp.Expression] | None:
-            nonlocal changed
-
-            exp.replace_children(
-                node, lambda n: n if isinstance(n, exp.Lambda) else evaluate_macros(n)
-            )
             if isinstance(node, MacroFunc):
                 changed = True
                 return self.evaluate(node)
             return node
 
-        transformed = evaluate_macros(expression)
+        transformed = evaluate_macros(expression.copy())
 
         if changed:
             # the transformations could have corrupted the ast, turning this into sql and reparsing ensures
             # that the ast is correct
             if isinstance(transformed, list):
-                return [self.parse_one(node.sql(dialect=self.dialect)) for node in transformed]
+                return [
+                    self.parse_one(node.sql(dialect=self.dialect, copy=False))
```

Comment:
what does `copy=False` mean for the `sql` call?

Reply:
when we generate sql using sqlglot.sql(), it by default does a copy for safety, sometimes write sql in a dialect has to do transformations. in this case, i know it's safe to copy=False here even though the sql generation may cause mutation.